Update darkmode functionality

Sirmorrison opened the pull request:
This pr updates darkmode feature.
fix #568

Sirmorrison committed:
move colors to values package

Sirmorrison committed:
update text colors

Sirmorrison committed:
Update icon color
- update Line/border color

Sirmorrison committed:
Fix issues with dark mode toggle on/off functionality
* Fix toggle button dark mode colour
* Clean up list preference modal
* Update list preference on settings page
* Move currency map to values package
* Clean up code

itswisdomagain committed:
stylable switch and iconbutton widgets

The Switch and IconButton widgets are modified to use color styles that
are tied to the page theme, causing the colors to update when the theme
is changed.

itswisdomagain committed:
implement dynamic styling for clickables and clickable lists

Sirmorrison committed:
Rebase pr with style fix
- Fix start page text dark mode color
- Remove hard coded net type.
- Remove excess scroll right padding

Sirmorrison committed:
Fix radio button colours
Fix editor text size bug
Fix checkbox icon and text color.
Fix scroll list color
Fix send amount editor line color

Sirmorrison committed:
Update darkmode icons on wallet,
- staking, account selector, vsp selector,
proposal legend, debug page, about page etc.

Sirmorrison committed:
fix lint error

devchoplife commented on ui/page/send/utxo_page.go:
```diff
@@ -84,7 +84,7 @@ func (pg *UTXOPage) Handle() {
 			icoBtn := pg.Theme.IconButton(decredmaterial.MustIcon(widget.NewIcon(icons.ContentContentCopy)))
 			icoBtn.Inset, icoBtn.Size = layout.UniformInset(values.MarginPadding5), values.MarginPadding20
-			icoBtn.Background = pg.Theme.Color.LightGray
+			icoBtn.ChangeColorStyle(&values.ColorStyle{Background: pg.Theme.Color.Gray4}) // TODO: Set foreground color too?
```
Are we setting foreground color here??

devchoplife commented on ui/util.go:
```diff
@@ -38,10 +37,6 @@ func editorsNotEmpty(editors ...*widget.Editor) bool {
 	return true
 }
 
-func generateRandomNumber() int {
-	return rand.New(rand.NewSource(time.Now().UnixNano())).Int()
-}
```
Remove GenerateRandomNumber function on ui/page/utils.go too. 
Here https://github.com/planetdecred/godcr/blob/01988a89c4ed254cd1403a64e0f8fcd95443881d/ui/page/utils.go#L40

Sirmorrison committed:
remove redundant code and comment

devchoplife reviewed (dismissed):
Looks good and works fine across all pages

dreacot reviewed (commented):
Stellar work was done on this PR, there are certain dark mode issues that still persist, i'm of the opinion that this PR gets merged regardless and issues found are broken down and tackled in bits in order not to unnecessarily delay having these changes in master

1. The specific peer text needs adjustment
<img width="567" alt="Screenshot 2021-12-06 at 8 37 11 PM" src="https://user-images.githubusercontent.com/25265396/144910828-988e587a-f350-4e59-bb70-8dd8abd42df0.png">

2. the wallet expand icon, and the arrow icon on the backup card needs to be modified
<img width="592" alt="Screenshot 2021-12-06 at 8 39 55 PM" src="https://user-images.githubusercontent.com/25265396/144911252-c39bf1a0-0ae0-4e42-a732-4964faee303f.png">

3. the rewards earned value isn't visible
<img width="603" alt="Screenshot 2021-12-06 at 8 41 47 PM" src="https://user-images.githubusercontent.com/25265396/144911498-e9630e2d-7d33-456b-b767-25523df2c9ef.png">

4. the expired icon needs to be modified
<img width="616" alt="Screenshot 2021-12-06 at 8 42 10 PM" src="https://user-images.githubusercontent.com/25265396/144911549-a0b1dbd8-1dd6-4d61-aea8-d95280ae4e5d.png">

5. the hover colour needs to be adjusted
<img width="599" alt="Screenshot 2021-12-06 at 8 42 42 PM" src="https://user-images.githubusercontent.com/25265396/144911621-c3d6af39-2e1b-4537-ad05-f6844479db2d.png">

6. the refresh icon on the proposals page needs to be adjusted
<img width="372" alt="Screenshot 2021-12-06 at 8 43 15 PM" src="https://user-images.githubusercontent.com/25265396/144911687-e7e37d96-edbf-48ab-a65a-64cbcfc2bfa3.png">

7. the view on politeia icon needs to be modified
<img width="602" alt="Screenshot 2021-12-06 at 8 43 34 PM" src="https://user-images.githubusercontent.com/25265396/144911732-070d1088-0310-4590-8238-43bae5e7fe7f.png">

8. hovering on the scrollbar produces a colour that's too dark
<img width="379" alt="Screenshot 2021-12-06 at 8 48 36 PM" src="https://user-images.githubusercontent.com/25265396/144912414-6c58252c-0438-4f13-bb88-894759caac5f.png">

9. the max buttons need to be consistent, one has a white text, one has a black text, also, the arrow in-between both input boxes need work
<img width="560" alt="Screenshot 2021-12-06 at 8 49 39 PM" src="https://user-images.githubusercontent.com/25265396/144912544-3ac1fc63-8f7b-4487-a413-190bb1e315ab.png">

10.  the icons on the more page needs to be modifed
<img width="268" alt="Screenshot 2021-12-06 at 8 50 13 PM" src="https://user-images.githubusercontent.com/25265396/144912618-db611919-a70e-4129-b182-b57914125d94.png">

11. this modal needs work, firstly the input box for the number of tickets should be a different shade of black
<img width="404" alt="Screenshot 2021-12-06 at 8 51 06 PM" src="https://user-images.githubusercontent.com/25265396/144912738-2c5719fd-a40b-448b-96c3-3b7e5ee35efb.png">

there are other minor issues like this lurking around

dreacot commented on ui/page/start_page.go:
```diff
@@ -178,14 +167,26 @@ func (sp *startPage) loadingSection(gtx layout.Context) layout.Dimensions {
 				layout.Rigid(func(gtx layout.Context) layout.Dimensions {
-					return sp.networkType.Layout(gtx)
+					netType := sp.WL.Wallet.Net
+					if sp.WL.Wallet.Net == "testnet3" {
```
`testnet3` this should be a constant as it's used in more than one place

Sirmorrison replied:
The display text should be `testnet` not `testnet3` as `testnet3` is the name of the backend folder.

dreacot replied:
I meant to say the if statement should check for something like `if sp.WL.Wallet.Net == Constants.TESNET3`

Sirmorrison committed:
Fix user agent and peer font color
update darkmode chevron icon on collapsible widget
update seed backup arrorw color to mockup color
update stake reward section
- Fix text color (darkmode and light mode)
- Align section content to middle
- Add no stake reward label
Update list scrollbar hover color
update usd max button text color
Replace testnet3 with dcrlibwallet.Testnet3 constant
